Composer/packages/client/src/pages/publish/Publish.tsx

```diff
@@ -210,24 +226,20 @@ const Publish: React.FC<RouteComponentProps<{ projectId: string; targetName?: st
     }
   }, [projectId]);
 
+  const prevBotStatusList: IBotStatus[] | undefined = usePrevious(botStatusList);
   useEffect(() => {
-    // init publishHistoryList
-    if (Object.keys(publishHistoryList).length > 0) {
+    // init bot status list for the botProjectData is empty array when first mounted
+    if (!isEqual(prevBotStatusList, statusList)) {
```

Comment:
I am having a hard time understanding why this is necessary.

This is only used to compare the status list and then update if changed. Can that not be done using the current state value?

```ts
if (!isEqual(currentStatusList, statusList) {
  setCurrentStatusList(statusList)
}
```

The `usePrevious` hook updates the ref in an effect, which is async anyways so this value could be stale.

Reply:
Remove `useRef` hook.